Uppgift12-Garage20/Controllers/ParkedVehiclesController.cs

```diff
@@ -32,12 +32,41 @@ private async Task<bool> IsRegistrationNumberUnique(string registrationNumber)
             return parkedVehicle == null;
         }
 
-        // GET: ParkedVehicles
-        public async Task<IActionResult> Index()
+        // Retrieves a list of parked vehicles based on the specified sorting order and search string.
+        public async Task<IActionResult> Index(string sortOrder, string searchString)
         {
-            return View(await _context.ParkedVehicle
-                .Select(vehicle => new VehicleSummaryViewModel(vehicle))
-                .ToListAsync());
+            ViewData["CurrentFilter"] = searchString;
+            ViewData["VehicleTypeSortParam"] = String.IsNullOrEmpty(sortOrder) ? "type_desc" : "type_asc";
```

Comment:
```suggestion
            ViewData["VehicleTypeSortParam"] = sortOrder == "type_asc" ? "type_desc" : "type_asc";
```

This should fix issue when toggling Vehicle Type filter between "asc" and "desc" sort.